deepchem/feat/deepvariant_featurizer.py

```diff
@@ -56,125 +57,106 @@ def left_align_indel(self, seq: str, pos: int,
         allele at each position, updating the allele counts dictionary.
 
         Parameters
         ----------
-
-        pileup_info : List[Dict[str, Any]]
-            List of dictionaries containing pileup information.
+        input_file : str
+            Path to the input BAM file.
         reference_seq_dict : Dict[str, str]
             Dictionary with chromosome names as keys and reference
             sequences as values.
-        allele_counts : Dict[Tuple[str, int], Dict[str, Any]]
-            Dictionary to store allele counts.
+
+        Returns
+        -------
+        Dict[Tuple[str, int], Dict[str, Any]]
+            Dictionary of allele counts.
 
         """
-        for pileupcolumn in pileup_info:
-            ref_base = reference_seq_dict[pileupcolumn['name']][
-                pileupcolumn['pos']]
+        allele_counts: Dict[Tuple[str, int], Dict[str, Any]] = {}
+        datapoint = pysam.AlignmentFile(input_file, "rb")
+        for pileup_item in datapoint.pileup():
+            pileupcolumn = cast(PileupColumn, pileup_item)
+            ref_pos = pileupcolumn.reference_pos
+            ref_name = pileupcolumn.reference_name
+            # Skip if reference sequence is not available
+            if ref_name not in reference_seq_dict:
+                continue
+
+            # Skip if position is out of bounds
+            if ref_pos >= len(reference_seq_dict[ref_name]):
+                continue
+
+            ref_base: str = reference_seq_dict[ref_name][ref_pos]
+
             allele_count: Dict[str, Any] = {
                 "reference_base": ref_base,
                 "read_alleles": defaultdict(int),
                 "coverage": 0
             }
-            for read_seq in pileupcolumn['reads']:
-                if read_seq[2]:
+            for pileup_read_item in pileupcolumn.pileups:
+                pileupread = cast(PileupRead, pileup_read_item)
+                read_info: List[Optional[Any]] = [
+                    pileupread.alignment.query_sequence,
+                    pileupread.query_position, pileupread.is_del,
+                    pileupread.is_refskip, pileupread.indel
+                ]
+                if read_info[2]:  # is_del
                     base = '-'
-                    print(allele_count)
                     allele_count["read_alleles"][base] += 1
                     allele_count["coverage"] += 1
-                elif read_seq[3]:
+                elif read_info[3]:  # is_refskip
                     base = 'N'
                     allele_count["read_alleles"][base] += 1
                     allele_count["coverage"] += 1
                 else:
                     indel = ""
-                    if read_seq[4] > 0:
-                        indel_seq = read_seq[0][read_seq[2]:read_seq[2] +
-                                                read_seq[4]]
-                        indel = f"+{read_seq[4]}{indel_seq}"
+                    if read_info[4] is not None and read_info[
```

Comment:
Can you add more comments here? This algorithm is a little complex and it will be hard to follow